src/auth/auth.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@nestjs/common';
+import { UserService } from '../user/user.service';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly usersService: UserService,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  async handleGoogleSignin(userInfo: { email: string; imageUrl: string }) {
+    const { email, imageUrl } = userInfo;
+
+    // Find or create the user
+    const user = await this.usersService.findOrCreateUser({ email, imageUrl });
```

Comment:
- First have a method to fetch user information from google using the provided token
- Check if the user exists using the email address we get from google 
- if user exists -> Register